source_control/gh_keys.py

```diff
@@ -0,0 +1,197 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2013, André Paramés <git@andreparames.com>
+# Based on the Git module by Leonardo Comelli <leonardo.comelli@gmail.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: gh_keys
+version_added: 1.9 
+short_description: Manages github ssh keys.
+description:
+    - The module manages the ssh key for a specific user through Github API v3.
+
+options:
+  action:
+    description: This tells the gh_keys module what you want it to do.
+    choices: [ present, absent ]
+
+  user:
+    description: Github username.
+    required: true
+
+  password:
+    description: Github password. If 2FA is enabled for your account, you should generate a new personal access token. Required for get_key, add_key and remove_key 
+    required: false 
```

Comment:
If required is false, you need to document the default, even if it's ‘None’ -- applies to password, title, key and key_id